[Feature] module unification controller & controller-test blueprints

dexturr opened the pull request:
Part of ember-cli/ember-cli#7530

Module unification blueprints for controller and controller-test files. This was pair programmed by @kevinansfield and I at the Ember London Project Night.

Many thanks to @GavinJoyce for showing us what to do and walking us through module unification.

GavinJoyce commented on package-lock.json:
```diff
@@ -0,0 +1,13015 @@
+{
```
can you delete this file?

dexturr replied:
Removed 👍

GavinJoyce commented on blueprints/controller/module-unification-files/__root__/__path__/controller.js:
```diff
@@ -0,0 +1,4 @@
+import Controller from '@ember/controller';
+
+export default Controller.extend({
+});
```
this looks like the same content as the existing https://github.com/dexturr/ember.js/blob/d23b14d99095c062693193e447b0ca578006e063/blueprints/controller/files/__root__/__path__/__name__.js. Perhaps we can use that for both and delete this file?

dexturr replied:
Yep, was copying the pattern of components but this is unnecessary here. Have done this now 👍

GavinJoyce commented on node-tests/blueprints/controller-test-test.js:
```diff
@@ -92,4 +157,18 @@ describe('Blueprint: controller-test', function() {
+        );
+      });
+    });
+  });
```
perhaps it would be good to add a test for a nested controller tests for addons and apps for both MU and classic? we seem to be missing those test cases

GavinJoyce commented:
This is great! Nice one!

I've added one final comment. Perhaps you might also squash down to a single commit when you can?

Commit:
[Feature] module unification controller & controller-test blueprints


Remove missed comment


Remove packagelock.json


Remove uneeded file are per PR comments


Add tests for nested components

GavinJoyce commented:
CI failure seems to be flakiness:

<img width="971" alt="screen shot 2018-05-18 at 15 43 23" src="https://user-images.githubusercontent.com/2526/40241175-3ebfc374-5ab2-11e8-9c9a-5bc89b6745e1.png">

dexturr commented:
Am I able to rerun the build? I couldn't seem to find the option or do I need to make a whitespace commit?

GavinJoyce commented:
👍 great work @dexturr @kevinansfield 👏
